dataland-frontend/tests/e2e/specs/search-company-framework-data/SearchCompaniesForFrameworkDataDropdownFilter.ts

```diff
@@ -103,36 +104,49 @@ describe("As a user, I expect the search functionality on the /companies page to
     "Checks that the sector filter synchronises between the search bar and the drop down and works",
     { scrollBehavior: false },
     () => {
-      const demoCompanyToTestFor = companiesWithEuTaxonomyDataForNonFinancials[0].companyInformation;
+      const demoCompanyToTestFor = assertDefined(
+        companiesWithEuTaxonomyDataForNonFinancials.find((it) => it.companyInformation.sector !== undefined)
+      ).companyInformation;
       const demoCompanyWithDifferentSector = companiesWithEuTaxonomyDataForNonFinancials.find(
-        (it) => it.companyInformation.sector !== demoCompanyToTestFor.sector
+        (it) =>
+          it.companyInformation.sector !== demoCompanyToTestFor.sector && it.companyInformation.sector !== undefined
       )!.companyInformation;
 
+      expect(demoCompanyToTestFor.sector).to.not.be.undefined;
       cy.ensureLoggedIn();
       cy.intercept("**/api/companies/meta-information").as("companies-meta-information");
-      cy.visit(`/companies?input=${demoCompanyToTestFor.companyName}&sector=${demoCompanyWithDifferentSector.sector}`)
+      cy.visit(`/companies?input=${demoCompanyToTestFor.companyName}&sector=${demoCompanyWithDifferentSector.sector!}`)
         .wait("@companies-meta-information")
         .get("div[class='col-12 text-left']")
         .should("contain.text", failureMessageOnAvailableDatasetsPage)
         .get("#sector-filter")
         .click()
         .get('input[placeholder="Search sectors"]')
-        .type(`${demoCompanyToTestFor.sector}`)
+        .type(`${demoCompanyToTestFor.sector!}`)
         .get("li")
-        .contains(RegExp(`^${demoCompanyToTestFor.sector}$`))
+        .contains(RegExp(`^${demoCompanyToTestFor.sector!}$`))
         .click()
         .get("td[class='d-bg-white w-3 d-datatable-column-left']")
         .contains(demoCompanyToTestFor.companyName)
-        .should("exist")
-        .url()
-        .should("contain", `sector=${convertStringToQueryParamFormat(demoCompanyToTestFor.sector)}`);
+        .should("exist");
+      cy.url().then((url) => {
+        if (demoCompanyToTestFor.sector) {
+          expect(url).to.contain(`sector=${convertStringToQueryParamFormat(demoCompanyToTestFor.sector)}`);
+        } else {
+          expect(url).to.not.contain("sector=");
+        }
+      });
```

Comment:
Das `if` und `else` ist doch eigentlich nicht nötig oder?
Es wurde bereits zweimal sichergestellt, dass der `sector` nicht undefined ist.
Daher würde ich hier nur noch 
`expect(url).to.contain(`sector=${convertStringToQueryParamFormat(demoCompanyToTestFor.sector)}`);`
machen.

Reply:
Gut gesehen. Das war vor dem expect Test da. To be pushed.